frontend/packages/ceph-storage-plugin/src/components/modals/clone-pvc-modal/clone-pvc-modal.tsx

```diff
@@ -0,0 +1,193 @@
+        throw error;
+      });
+  };
+
+  const updateClonePVCName = (value: string) => {
+    setClonePVCName(value);
+  };
+
+  return (
+    <Form onSubmit={submit}>
+      <div className="modal-content modal-content--no-inner-scroll">
+        <ModalTitle>Clone</ModalTitle>
+        <ModalBody>
+          <FormGroup
+            label="Name"
+            isRequired
+            fieldId="clone-pvc-modal__name"
+            className="co-required"
+          >
+            <TextInput
+              isRequired
+              type="text"
+              id="clone-pvc-modal__name"
+              name="clone-pvc-modal__name"
+              value={clonePVCName}
+              onChange={updateClonePVCName}
+            />
+          </FormGroup>
+          <div className="clone-pvc-modal__details-section">
+            <p className="text-muted clone-pvc-modal__pvc-details">PVC Details</p>
+            <div className="clone-pvc-modal__details">
+              <div>
+                <div>
+                  <p className="clone-pvc-modal__pvc-details">Namespace</p>
+                  <p>
+                    <ResourceIcon kind="Namespace" />
+                    {resource.metadata.namespace}
+                  </p>
+                </div>
+                <div>
+                  <p className="clone-pvc-modal__pvc-details">Storage Class</p>
+                  <p>
+                    <ResourceIcon kind="StorageClass" />
+                    {resource.spec.storageClassName}
+                  </p>
+                </div>
+              </div>
+              <div>
+                <div>
+                  <p className="clone-pvc-modal__pvc-details">Requested Capacity</p>
+                  <p>
+                    {
+                      humanizeBinaryBytes(
+                        convertToBaseValue(resource.spec.resources.requests.storage),
+                      ).string
+                    }
+                  </p>
+                </div>
+                <div>
+                  <p className="clone-pvc-modal__pvc-details">Used Capacity</p>
+                  <p>{!queriesLoadError && humanizeBinaryBytes(pvcUsedCapacity).string}</p>
+                </div>
+              </div>
+              <div>
+                <div>
+                  <p className="clone-pvc-modal__pvc-details">Access Mode</p>
+                  <p>{accessModeRadios[resource.spec.accessModes]}</p>
+                </div>
+                <div>
+                  <p className="clone-pvc-modal__pvc-details">Volume Mode</p>
+                  <p>{resource.spec.volumeMode}</p>
+                </div>
+              </div>
+            </div>
+          </div>
+        </ModalBody>
+        <ModalSubmitFooter
+          inProgress={inProgress}
+          errorMessage={errorMessage}
+          submitText="Clone"
+          cancel={cancel}
+          submitDisabled={buttonDisabled}
+        />
+      </div>
+    </Form>
+  );
+});
+
+export type ClonePVCModalProps = {
+  kind?: any;
```

Comment:
```suggestion
  kind?: any;
```
why any? Isn't kind a string?